libopendavinci/src/odcore/base/module/ManagedClientModule.cpp

```
@@ -116,12 +121,116 @@ namespace odcore {
                 return m_startOfLastCycle;
             }
 
+            std::vector<double> ManagedClientModule::computeAdvancedStatistics() {
+#ifdef __linux__
+                bool success=false;
+                double cpu_stats[4];
+                string NO_SYS_INFO_EXC_MSG="Could not get system info";
+                
+                try
+                {
+                    // get system info
+                    
+                    // The number of processors currently online
+                    int numCPU = sysconf(_SC_NPROCESSORS_ONLN);
+                    if(numCPU == -1) {
+                        OPENDAVINCI_CORE_THROW_EXCEPTION(IOException,NO_SYS_INFO_EXC_MSG);
```

We need to be extract careful in this class as all modules depend on its flawless behavior.

I suggest to avoid throwing an exception here but adjust the code in such a way that it can handle ::sysconf(...) calls that might fail or return unexpected values without throwing an exception. @fgiaimo Could you please adjust the code accordingly? Thanks.

---

Would a silent fail be better? The code can send out dummy values like -1 if there are errors

---

Or a boolean value is_valid can be added to the final statistics data structures (CPUConsumption and MEMConsumption)

---

The question is also if this call failed once, wouldn't it fail always? I.e., could you add a flag to avoid re-executing and failing to execute sysconf(...) repeatedly?